src/holdingpen/components/ExceptionsTable.jsx

```diff
@@ -0,0 +1,168 @@
+        focused: false,
+      },
+      error: {
+        filterDropdownVisible: false,
+        searchText: '',
+        filtered: false,
+        focused: false,
+      },
+    };
+  }
+
+  onFilterDropdownVisibleChange(visible, key) {
+    const keyToUpdate = { ...this.state[key] };
+    keyToUpdate.filterDropdownVisible = visible;
+    keyToUpdate.focused = visible;
+    this.setState({
+      [key]: keyToUpdate,
+    });
+  }
+
+  onInputChange(e, key) {
+    const keyToUpdate = { ...this.state[key] };
+    keyToUpdate.searchText = e.target.value;
+    this.setState({
+      [key]: keyToUpdate,
+    })
+  }
+
+  onSearch(key) {
+    const exceptions = [...this.props.exceptions];
+    const keyToUpdate = { ...this.state[key] };
+    const { searchText } = keyToUpdate;
+    const reg = new RegExp(searchText, 'gi');
+    keyToUpdate.filterDropdownVisible = false;
+    keyToUpdate.filtered = !!searchText;
+    this.setState({
+      [key]: keyToUpdate,
+      exceptions: exceptions.map((exception) => {
+        const match = exception[key].match(reg);
+        if (!match) {
+          return null;
+        }
+        return {
+          ...exception,
+          [key]: (
+            <span>
+              {exception[key].split(new RegExp(`(?<=${searchText})|(?=${searchText})`, 'i')).map((text, i) => (
+                text.toLowerCase() === searchText.toLowerCase() ?
+                  <span key={i} className="highlight">{text}</span> : text // eslint-disable-line
+              ))}
+            </span>
+          ),
+        };
+      }).filter(exception => !!exception),
+    });
+  }
+
+  handleChange(pagination, filters) {
+    this.setState({
+      filteredInfo: filters,
+    });
+  }
+
+  clearFilters() {
+    this.setState({ filteredInfo: null });
+  }
+
+  renderRecords(text) {
+    if (typeof text === 'string') {
+      const recordLinks = text.split('\n');
+      return recordLinks.map((recordLink, index) => <p key={index}>{recordLink}</p>);
+    } else {
+      return text;
+    }
+  }
+
+  render() {
+    const filteredInfo = this.state.filteredInfo || {};
+
+    const columns = [{
+      title: 'Collection',
+      dataIndex: 'collection',
+      key: 'collection',
+      filters: [
+        { text: 'HEP', value: 'HEP' },
+        { text: 'HEPNAMES', value: 'HEPNAMES' },
+        { text: 'JOB', value: 'JOB' },
+        { text: 'JOBHIDDEN', value: 'JOBHIDDEN' },
+        { text: 'CONFERENCES', value: 'CONFERENCES' },
+        { text: 'JOURNALS', value: 'JOURNALS' },
+      ],
+      filteredValue: filteredInfo.collection,
+      onFilter: (value, record) => record.collection.includes(value),
+      width: 120,
+    },
+    {
+      title: 'Error',
+      dataIndex: 'error',
+      key: 'error',
+      filterDropdown: <FilterDropdown
+        placeholder="Search error"
+        filterByKey="error"
+        searchText={this.state.error.searchText}
+        onInputChange={event => this.onInputChange(event, 'error')}
```

Comment:
remove this, and pass search text in `onSearch` it is when you perform search anyhow.

Reply:
also in general, don't pass event objects unless it is necessary, just pass what you parent exactly needs.